monitoring/backup_stats_number_failed_backups_metric.go

```diff
@@ -0,0 +1,47 @@
+package monitoring
+
+import (
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type numberFailedBackupsMetric struct {
+	Desc *prometheus.Desc
+	Type prometheus.ValueType
+}
+
+func newNumberFailedBackupsMetric() numberFailedBackupsMetric {
+	return numberFailedBackupsMetric{
+		Desc: prometheus.NewDesc(
+			prometheus.BuildFQName(longhornName, subsystemBackupStats, "number_failed_backups"),
+			"Total number of failed backups of this volume",
+			[]string{prometheusVolumeIDLabel},
+			nil,
+		),
+		Type: prometheus.CounterValue,
+	}
+}
+
+func (m numberFailedBackupsMetric) describe() *prometheus.Desc {
+	return m.Desc
+}
+
+func (m numberFailedBackupsMetric) getMetricValues(ch chan<- prometheus.Metric, c *Collector) (err error) {
+	defer func() {
+		err = errors.Wrap(err, "failed to get values for numberFailedBackupsMetric")
+	}()
+
+	volumeLists, err := c.lhClient.LonghornV1beta1().Volumes(c.namespace).List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	for _, v := range volumeLists.Items {
```

Comment:
Don't store these persistently, in the objects, this will create a lot of update and writes which are completely unnecessary. 

Metrics should only exist at runtime in memory => hashmap
Prometheus can collect these, from your metrics endpoints.

Reply:
applies to all the other metrics as well, also see implementation comment above for the increment methods etc.

Reply:
We will discuss more about this on the meeting tommorow